File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0101.java
//101. 对称二叉树
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0101 {

}
class Solution {
  public boolean compare(TreeNode left,TreeNode right){
    if(left==null&&right!=null) return false;
    else if(left!=null&&right==null) return false;
    else if(left==null&&right==null) return true;
    else if(left.val!=right.val) return false;
    else return compare(left.left, right.right)&&compare(left.right, right.left);
  }
  public boolean isSymmetric(TreeNode root) {
    if(root==null) return true;
    else return compare(root.left,root.right);
  }
}

File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0101.go
//101. 对称二叉树
package main

import (
	. "binaryTree/types"
)

func compare(left,right *TreeNode) bool{
	if left!=nil&&right==nil {
		return false
	}else if left==nil&&right!=nil {
		return false
	}else if left==nil&&right==nil {
		return true
	}else if left.Val!=right.Val {
		return false
	}else{
		return compare(left.Left,right.Right)&&compare(left.Right,right.Left)
	}
}

func isSymmetric(root *TreeNode) bool {
	if root==nil {
		return true
	}else{
		return compare(root.Left,root.Right)
	}
}
